src/doc/rustdoc/src/advanced-features.md

```diff
@@ -47,16 +47,15 @@ all type errors and name resolution errors with function bodies. Note that this
 work for anything outside a function body: since Rustdoc documents your types, it has to
 know what those types are! For example, this code will work regardless of the platform:
 
-<!-- `ignore` because doc-tests are run with `rustc`, not `rustdoc` -->
-```ignore
+```rust,ignore (platform-specific)
```

Comment:
This doesn't preserve the info from the comment IMO. The issue is that - regardless of whether rustc gives an error or not - it may not match rustdoc's behavior, so it would be confusing to show it.
```suggestion
```rust,ignore (tests rustdoc, not rustc)
```

Reply:
I don't think that is a good explanation of why it is ignored.  If it was just about the different output, then `noplayground` would be a more appropriate marker.  

The important difference is that with `ignore`, there's nothing to validate that the example contains valid rust code.  With `noplayground`, it'll be validated, but won't show the play button (which I assume is what the concern is about).  Generally it is best to avoid `ignore` if at all possible.

That being said, I added an annotation for that.

Reply:
> . If it was just about the different output, then noplayground would be a more appropriate marker.

It's not about the output - there is code that rustdoc accepts that rustc doesn't. So if this is marked with `no_playground` a) it will break on Linux because it's tested with rustc and b) even if it didn't break it would be misleading, because rustdoc and rustc are fundamentally treating this code differently.

If it makes you feel better, ignore still validates that the syntax is valid, just not that the program typechecks.

Reply:
Sorry, I didn't mean to seem like I was contradicting you; I could have worded that better. I was just trying to explain the reasoning I had for changing it. I understand there is a subtle reasoning here.

`ignore` does not validate syntax for tests.  That validation is only run when building API documentation with `rustdoc` (and even then, I think it is only a warning).  When running `rustdoc --test` you can put any text in an `ignore` block and it is completely ignored, and that's how the books are tested.

Reply:
Ah ok, that makes sense.

> ignore does not validate syntax for tests.

Well, maybe it shouldn't, but currently it does: https://github.com/rust-lang/rust/issues/30032